tsfc/fem.py

```diff
@@ -319,9 +317,10 @@ def callback(key):
         table = ctx.tabulation_manager[key]
         if len(table.shape) == 1:
             # Cellwise constant
-            row = gem.Literal(table)
+            row = table
         else:
-            table = ctx.index_selector(lambda i: gem.Literal(table[i]), mt.restriction)
+            table = ctx.index_selector(lambda i: gem.partial_indexed(table, (i,)),
```

Comment:
So it's a `ListTensor` at this point that contains an array of `Indexed` objects where the `gem.Failure` is in. When tsfc gets to this stage, `table.array` of the `ListTensor` has the form: `array([[[ Indexed(Failure(shape, TraceError()), (0, 0)), Indexed(Failure(shape, TraceError()), (0, 1)), ... ]]], dtype=object)`

I managed to find where it goes wrong. It goes wrong in the constructor for `Literal` when it forces the array as `self.array = asarray(array, dtype=float)`. I get a `TypeError` here in the case the `ListTensor` contains indexed gem failures.